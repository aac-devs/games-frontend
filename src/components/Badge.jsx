import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { backgroundColor, textColor } from '../global-styles';

const BadgeContainer = styled.div`
  position: relative;
  display: flex;
  flex: 0 1 auto;
  align-items: center;
  max-width: 400px;
  box-sizing: border-box;
  margin-right: 12px;
  margin: 0 7px 7px 0;
  padding: 5px ${(props) => (props.hasMarginRight ? '30px' : '10px')} 5px 10px;
  border-radius: 20px;
  border: 1px solid hsla(0, 0%, 100%, 0.2);
  font-family: 'Roboto', sans-serif, Helvetica, Arial;
  background-color: ${(props) => props.background};
  transition: border-color 0.3s;
`;

const BadgeText = styled.div`
  font-size: 16x;
  text-overflow: ellipsis;
  white-space: nowrap;
  transition: color 0.3s;
  overflow: hidden;
  color: ${textColor.primary.light};
`;

const BadgeButton = styled.button`
  background-color: ${backgroundColor.secondary.dark};
  color: ${textColor.primary.light};
  border: none;
  padding: 2px 5px;
  border-radius: 50%;
  position: absolute;
  top: 3.5px;
  right: 4px;
  cursor: pointer;
  transition: all 0.2s;
  &:hover {
    background-color: ${backgroundColor.secondary.normal};
  }
`;

const Badge = ({ id, text, name, hasCloseButton, handleClose }) => {
  const backColor =
    name === 'genres'
      ? backgroundColor.primary.dark
      : backgroundColor.primary.normal;
  return (
    <BadgeContainer background={backColor} hasMarginRight={hasCloseButton}>
      <BadgeText id="text">{text}</BadgeText>
      {hasCloseButton && (
        <BadgeButton onClick={() => handleClose(id, name)}>
          <i className="fas fa-times" />
        </BadgeButton>
      )}
    </BadgeContainer>
  );
};

Badge.propTypes = {
  id: PropTypes.number,
  text: PropTypes.string.isRequired,
  name: PropTypes.string,
  hasCloseButton: PropTypes.bool,
  handleClose: PropTypes.func,
};

Badge.defaultProps = {
  id: 1,
  name: 'genres',
  hasCloseButton: false,
  handleClose: () => {},
};

export default Badge;
