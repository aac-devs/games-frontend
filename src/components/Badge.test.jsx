/* eslint-disable no-console */
/* eslint-disable no-undef */
// import '@testing-library/jest-dom';
// import { render } from '@testing-library/react';

import React from 'react';
import { shallow } from 'enzyme';
import Badge from './Badge';

describe('Pruebas en el componente <Badge />', () => {
  // test('debe mostrar el mensaje "PlayStation 4"', () => {
  //   const message = 'PlayStation 4';
  //   const { getByText } = render(<Badge text={message} name="platforms" />);
  //   expect(getByText(message)).toBeInTheDocument();
  // });
  test('debe mostrar <Badge /> correctamente', () => {
    const platformText = 'PlayStation 4';
    const wrapper = shallow(<Badge text={platformText} name="platforms" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe mostrar el name enviado por props', () => {
    const platformText = 'PlayStation 4';
    const wrapper = shallow(<Badge text={platformText} name="platforms" />);
    const readedText = wrapper.find('#text').text();
    expect(readedText).toBe(platformText);
  });
  // TODO: continua en el video 61. Configurar el dotenv para los diferentes entornos.
});
