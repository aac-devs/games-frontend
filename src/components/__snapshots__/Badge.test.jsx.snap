// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <Badge /> debe mostrar <Badge /> correctamente 1`] = `
<styled.div
  background="#212121"
  hasMarginRight={false}
>
  <styled.div
    id="text"
  >
    PlayStation 4
  </styled.div>
</styled.div>
`;
